domain/network/service/space.go

```diff
@@ -136,22 +138,21 @@ func (s *Service) SaveProviderSubnets(
 				continue
 			}
 			fanSubnetID := generateFanSubnetID(subnetNet.String(), subnet.ProviderId.String())
-			if overlaySegment != nil {
```

Comment:
This was a tautological lint error. We already check for `overlaySegment == nil` just above the continue in the diff.